src/utils/sorting-functions.ts

```diff
@@ -0,0 +1,123 @@
+import { ToDoItem, Priority } from '../models/todoitem-model';
+
+export function sortByTitle(items: ToDoItem[], order: number): ToDoItem[] {
+  const orderedItems = items;
+  if (order === 1) {
+    orderedItems.sort((a, b) => {
+      const titleA = a.title.toLowerCase();
+      const titleB = b.title.toLowerCase();
+
+      if (titleA < titleB) {
+        return -1;
+      }
+      if (titleA > titleB) {
+        return 1;
+      }
+      return 0;
+    });
+  } else {
+    orderedItems.sort((a, b) => {
+      const titleA = a.title.toLowerCase();
+      const titleB = b.title.toLowerCase();
+
+      if (titleA > titleB) {
+        return -1;
+      }
+      if (titleA < titleB) {
+        return 1;
+      }
+      return 0;
+    });
+  }
+  return orderedItems;
+}
```

Comment:
Can you have a common high order sorting function neither having this if else in copy pasted everywhere
Try something like this:
`function sortBySpecificKey(key: keyof ToDoItem, order: number) {
  return function (a: ToDoItem, b: ToDoItem) {
    const titleA = a[key].toLowerCase();
    const titleB = b[key].toLowerCase();
    return titleA > titleB ? order : titleA < titleB ? -order : 0
  }
}`

You can use: 
`function sortByTitle(items: ToDoItem[], order: number): ToDoItem[] {
	return items.sort(sortBySpecificKey('title', order));
}`